neqo-transport/src/tracking.rs

```diff
@@ -453,11 +453,8 @@ impl RecvdPackets {
             return;
         }
 
-        builder.encode_varint(if self.ecn_count.is_some() {
-            FrameType::AckEcn
-        } else {
-            FrameType::Ack
-        });
+        // TODO: Send Ack and not AckEcn when no ecn counts available.
+        builder.encode_varint(FrameType::AckEcn);
```

Comment:
This change is a hack.

With this pull request we no longer need to send `FrameType::AckEcn` during the connection handshake, as the other side doesn't mark their datagrams during the handshake.

A QUIC AckECN is larger than a QUIC Ack, as it carries the various ECN counts.

The above size difference causes the server initial to fit into a single flight, thus breaking a lot of [`handshake.rs`](https://github.com/mozilla/neqo/blob/main/neqo-transport/src/connection/tests/handshake.rs) tests, specifically those that assume that the server initial doesn't fit in one flight and is thus later on coalesced with the handshake space.

Before I attempt fixing these, which is quite involved, double checking that this is the way we want to go.

@larseggert objections?